src/main/scala/it/pagopa/interop/purposeprocess/api/impl/OrganizationRole.scala

```diff
@@ -0,0 +1,22 @@
+package it.pagopa.interop.purposeprocess.api.impl
+
+import it.pagopa.interop.purposeprocess.error.InternalErrors.OrganizationNotAllowed
+
+import java.util.UUID
+import scala.concurrent.Future
+
+sealed trait OrganizationRole
+
+object OrganizationRole {
+
+  final case object CONSUMER           extends OrganizationRole
+  final case object PRODUCER           extends OrganizationRole
+  final case object PRODUCER_OF_ITSELF extends OrganizationRole
```

Comment:
I'll be honest, I dislike `PRODUCER_OF_ITSELF`, but no better idea except `CONSUMER_PRODUCER`.
Any suggestions?